juju/juju
# Feature/caas operator use svc ip

## Description of change

changes made for supporting `upgrade-[controller|model]` for k8s controller:
`apiaddresses` in agent config for a `CAAS` operator set to `CloudService.Addresses` for a k8s controller;

## QA steps

1. bootstrap into microk8s

```bash
$ juju bootstrap microk8s k1 --debug 
```

2. add model then deploy `CAAS` workloads;

```bash
$ juju add-model t1 microk8s && \
juju create-storage-pool mariadb-pv kubernetes storage-class=microk8s-hostpath && \
juju create-storage-pool operator-storage kubernetes storage-class=microk8s-hostpath && \
juju deploy cs:~juju/mariadb-k8s-0  --debug --storage database=500M,mariadb-pv && \
juju deploy cs:~juju/gitlab-k8s-0  --debug  --config juju-external-hostname=127.0.0.1.xip.io && \
juju relate mariadb-k8s gitlab-k8s && \
juju expose gitlab-k8s
```

3. verify `apiaddresses` in `agent.conf`;

```bash
$ kubectl exec gitlab-k8s-operator-0  -n t1 -it cat /var/lib/juju/agents/application-gitlab-k8s/agent.conf
```


## Documentation changes

None;

## Bug reference

Not ready;
